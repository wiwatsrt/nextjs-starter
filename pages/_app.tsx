import { AppProps } from 'next/app'
import { AuthProvider } from '../context/authContext'

import '../styles/tailwind.css'

export default function App({ Component, pageProps }: AppProps) {
  return (
    <AuthProvider>
      <Component {...pageProps} />
    </AuthProvider>
  )
}
