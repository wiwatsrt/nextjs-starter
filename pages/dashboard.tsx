import React from 'react'
import AppLayout from '../components/Layout/AppLayout'

interface Props {}

const dashboard = (props: Props) => {
  return (
    <AppLayout header="Dashboard">
      <div className="py-4">
        <div className="border-4 border-dashed border-gray-200 rounded-lg h-96"></div>
      </div>
    </AppLayout>
  )
}

export default dashboard
