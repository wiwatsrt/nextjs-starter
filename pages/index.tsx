import Head from 'next/head'

export const Home = (): JSX.Element => (
  <div className="antialiased text-gray-900 flex items-center justify-center min-h-screen">
    <Head>
      <title>Create Next App</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>
    <div>
      <h1 className="text-5xl font-bold text-purple-500">Hello world!</h1>
    </div>
  </div>
)

export default Home
