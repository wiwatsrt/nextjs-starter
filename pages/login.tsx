import Head from 'next/head'
import { useRouter } from 'next/router'
import React, { ChangeEvent, useEffect, useState } from 'react'
import { useAuth } from '../context/authContext'

const LoginPage = () => {
  const router = useRouter()
  const { auth, signInWithEmailAndPassword } = useAuth()

  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')

  useEffect(() => {
    if (auth) {
      router.push('/')
    }
  }, [auth])

  const handleSubmit = async (e: ChangeEvent<any>) => {
    e.preventDefault()
    try {
      await signInWithEmailAndPassword(email, password)
      router.push('/')
    } catch (error) {
      console.error(error)
    }
  }

  return (
    <div>
      <Head>
        <title>Login Page</title>
      </Head>
      <div className="min-h-screen flex items-center justify-center bg-gray-900 py-12 px-4 sm:px-6 lg:px-8">
        <div className="max-w-md w-full">
          <form
            onSubmit={handleSubmit}
            className="bg-white shadow rounded-lg p-8"
          >
            <h2 className="mb-6 uppercase text-center text-3xl leading-9 font-medium text-gray-900">
              Login
            </h2>
            <div className="mb-4">
              <label
                className="block mb-2 font-bold text-gray-700"
                htmlFor="email"
              >
                Email
              </label>
              <input
                id="email"
                onChange={(e) => setEmail(e.target.value)}
                className="form-input mt-1 block w-full"
                type="email"
                value={email}
                autoComplete="email"
                required
              />
            </div>

            <div className="mb-4">
              <label
                className="block mb-2 font-bold text-gray-700"
                htmlFor="password"
              >
                Password
              </label>
              <input
                id="password"
                onChange={(e) => setPassword(e.target.value)}
                className="form-input mt-1 block w-full"
                type="password"
                value={password}
                autoComplete="current-password"
                required
              />
            </div>

            <div className="mt-6 flex items-center justify-between">
              <div className="flex items-center">
                <input
                  id="remember_me"
                  v-model="form.rememberMe"
                  className="form-checkbox"
                  type="checkbox"
                />
                <label
                  htmlFor="remember_me"
                  className="ml-2 text-sm text-gray-900"
                >
                  Remember me
                </label>
              </div>

              <div className="text-sm">
                <a
                  href="#"
                  className="font-medium text-primary-600 hover:text-primary-500 focus:outline-none focus:underline"
                >
                  Forgot your password?
                </a>
              </div>
            </div>
            <div className="mt-6">
              <button
                type="submit"
                className="button button-primary w-full transition duration-150 ease-in-out"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  )
}

export default LoginPage
