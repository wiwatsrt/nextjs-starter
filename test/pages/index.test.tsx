import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders hello world', () => {
    const { getByText } = render(<Home />, {})
    expect(getByText('Hello world!'));
  })
})
