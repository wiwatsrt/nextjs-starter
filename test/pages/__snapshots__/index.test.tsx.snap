// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-da4d2267357886b1 container"
  >
    <main
      class="jsx-da4d2267357886b1"
    >
      <h1
        class="jsx-da4d2267357886b1 title"
      >
        Welcome to 
        <a
          class="jsx-da4d2267357886b1"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="jsx-da4d2267357886b1 description"
      >
        Get started by editing 
        <code
          class="jsx-da4d2267357886b1"
        >
          pages/index.tsx
        </code>
      </p>
      <button
        class="jsx-da4d2267357886b1"
      >
        Test Button
      </button>
      <div
        class="jsx-da4d2267357886b1 grid"
      >
        <a
          class="jsx-da4d2267357886b1 card"
          href="https://nextjs.org/docs"
        >
          <h3
            class="jsx-da4d2267357886b1"
          >
            Documentation →
          </h3>
          <p
            class="jsx-da4d2267357886b1"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="jsx-da4d2267357886b1 card"
          href="https://nextjs.org/learn"
        >
          <h3
            class="jsx-da4d2267357886b1"
          >
            Learn →
          </h3>
          <p
            class="jsx-da4d2267357886b1"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="jsx-da4d2267357886b1 card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="jsx-da4d2267357886b1"
          >
            Examples →
          </h3>
          <p
            class="jsx-da4d2267357886b1"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="jsx-da4d2267357886b1 card"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="jsx-da4d2267357886b1"
          >
            Deploy →
          </h3>
          <p
            class="jsx-da4d2267357886b1"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="jsx-da4d2267357886b1"
    >
      <a
        class="jsx-da4d2267357886b1"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="jsx-da4d2267357886b1 logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
